bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -785,14 +829,15 @@ public static Long win32_getHandle (Image image, int zoom) {
 		}
 		if (image.dataAtBaseZoom != null) {
 			ImageData resizedData = image.getImageData(zoom);
-			image.init(resizedData, zoom);
+			ImageData newData = image.copyImageDataForDisabledOrGrayImages(image.getDevice(), resizedData, image.styleFlag);
```

Comment:
I don't think it is neccessary to apply that here as well. As the image data is directly used from the existing image, I think it should already be greyed or disabled and only needs to be rescaled

Reply:
The imageData is not being directly used from the existing image (it is recalculated and not stored in as a variable). `getImageData()` method is calling `autoScaleImageData()` and it's returning a new instance of ImageData where we lose the gray/disabled property again. You can try it out by commenting the line I added.

Reply:
Ah yes, you are right